infra/aws/terraform/prow-build-cluster/README.md

```diff
@@ -243,3 +243,63 @@ make destroy
 
+
+## GitOps
+
+To sync state from this Git repo into EKS cluster we use [FluxCD](https://fluxcd.io/).
+
+FluxCD has a dedicated [CLI tool](https://fluxcd.io/flux/installation/#install-the-flux-cli). We use it iniside our scripts to generate and monitor syncs.
+
+FluxCD resources stored inside the `./resources` directory and have been generated with use of `./hack/flux-update.bash`. The script prepares manifests for the [GitOps Tool Kit](https://fluxcd.io/flux/components/) and generates [Sources](https://fluxcd.io/flux/components/source/), [Kustomizations](https://fluxcd.io/flux/components/kustomize/kustomization/) and [Helm Releases](https://fluxcd.io/flux/components/helm/helmreleases/).
+
+The `flux-system` namespace contains all GitOps Tool Kit componenets as well as all Flux Sources and Kustomizations. Helm Releases ought to be deployed in the same namespaces as manifests they create. As a convention, Flux Helm Releases have to be prefixed with `flux-hr-`, Kustomizations with `ks-` and sources should contain `-source-` part. These are our internal conventions and are used for discovery process in our scripts:
+
+- Installing Flux:
+    ```bash
+    make flux-install
+    ```
+
+- Deploying Sources:
+    ```bash
+    make flux-apply-sources
+    ```
+
+- Deploying Kustomizations:
+    ```bash
+    make flux-apply-kustomizations
+    ```
+
+- Deploying Helm Releases:
+    ```bash
+    make flux-apply-helm
+    ```
+
+### Accessing Flux
+
+In order to access FluxCD, all you need is kubeconfig with broad cluster permissions and Flux CLI installed locally.
+
+### Adding New Kustomization
+
+1. Go to `./resources` directory and create a new folder containing your manifests.
+2. Open `./hack/flux-update.bash` in your editor of choice.
+3. Find `kustomizations` variable. It contains a list of kustomizations to generate. Each element of list corresonds to directory inside `./resources` folder.
+4. Extend the kustomization list with the name of folder you created in the first step.
+5. Regenerate kustomizations by running: `make flux-update`
+6. Commit your changes and wait for your resources to appear in the cluster. You can use `flux get ks <kustomization-name>` to monitor state.
+
+### Adding New Helm Release
+
+1. Open `./hack/flux-update.bash` in your editor of choice.
+2. Create a new helm source. If you already have helm source, you can move to the next step.
+    1. Find part of script responsible for createing `eks-charts` helm release.
+    2. Use the same pattern to create a new helm source.
+3. Create a new helm release.
+    1. Find part of script responsible for creating helm release for `node-termination-handler`.
+    2. Based on `node-termination-handler` example extend the script with your helm release.
+4. Regenerate helm-releases by running: `make flux-update`
+5. Apply new helm release by running: `make flux-apply-helm-releases`
```

Comment:
Is this step actually needed? Isn't Flux automatically going to apply this to the cluster once committed since we have Kustomization for files in flux-system?

Reply:
You are sort of right. In order for you Helm Release to be applied automatically, it has to be part of some existing Kustomization. I'll rephrase this part.